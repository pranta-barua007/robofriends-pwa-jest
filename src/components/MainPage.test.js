import { shallow } from "enzyme";
import React from "react";
import MainPage from "./MainPage";

describe('MainPage', () => {

    let wrapper;
    beforeEach(() => {
        const mockProps = {
            onRequestRobots: jest.fn(),
            robots: [],
            searchField: '',
            isPending: false
        }
        wrapper = shallow(<MainPage {...mockProps} />)
    });
    
    it('should render ManinPage & match snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should filter robots correctly', () => {
        const mockProps2 = {
            onRequestRobots: jest.fn(),
            robots: [
                {
                    id: 5,
                    name: 'jon',
                    email: 'jon@gmail.com'
                }
            ],
            searchField: 'jon',
            isPending: false
        }
        const wrapper2 = shallow(<MainPage {...mockProps2}/>)
        expect(wrapper2.instance().filteredRobots()).toEqual([
            {
                id: 5,
                name: 'jon',
                email: 'jon@gmail.com'
            }
        ]);
    });

    it('should filter unmatched robots ', () => {
        const mockProps3 = {
            onRequestRobots: jest.fn(),
            robots: [
                {
                    id: 5,
                    name: 'jon',
                    email: 'jon@gmail.com'
                }
            ],
            searchField: 'pop',
            isPending: false
        }
        const wrapper3 = shallow(<MainPage {...mockProps3}/>)
        expect(wrapper3.instance().filteredRobots()).toEqual([]);
    });

});
