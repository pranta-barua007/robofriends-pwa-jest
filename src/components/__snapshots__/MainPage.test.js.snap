// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainPage should render ManinPage & match snapshot 1`] = `
<div
  className="tc"
>
  <Header />
  <SearchBox />
  <Scroll>
    <ErrorBoundary>
      <CardList
        robots={Array []}
      />
    </ErrorBoundary>
  </Scroll>
</div>
`;
